pkg/injector/envoy_config.go

```diff
@@ -152,60 +183,93 @@ func (wh *mutatingWebhook) createEnvoyBootstrapConfig(name, namespace, osmNamesp
+		UpstreamProtocolOptions: &xds_upstream_http.HttpProtocolOptions_ExplicitHttpConfig_{
+			ExplicitHttpConfig: &xds_upstream_http.HttpProtocolOptions_ExplicitHttpConfig{
+				ProtocolConfig: &xds_upstream_http.HttpProtocolOptions_ExplicitHttpConfig_Http2ProtocolOptions{},
 			},
 		},
-		"transport_socket": map[string]interface{}{
-			"name": "envoy.transport_sockets.tls",
-			"typed_config": map[string]interface{}{
-				"@type": "type.googleapis.com/envoy.extensions.transport_sockets.tls.v3.UpstreamTlsContext",
-				"common_tls_context": map[string]interface{}{
-					"alpn_protocols": []string{
-						"h2",
-					},
-					"validation_context": map[string]interface{}{
-						"trusted_ca": map[string]interface{}{
-							"inline_bytes": config.RootCert,
+	}
+	pbHTTPProtocolOptions, err := ptypes.MarshalAny(httpProtocolOptions)
+	if err != nil {
+		log.Error().Err(err).Msgf("Error marshaling HttpProtocolOptions struct into an anypb.Any message")
+		return nil, err
+	}
+
+	upstreamTLSContext := &xds_transport_sockets.UpstreamTlsContext{
+		CommonTlsContext: &xds_transport_sockets.CommonTlsContext{
+			AlpnProtocols: []string{
+				"h2",
+			},
+			ValidationContextType: &xds_transport_sockets.CommonTlsContext_ValidationContext{
+				ValidationContext: &xds_transport_sockets.CertificateValidationContext{
+					TrustedCa: &xds_core.DataSource{
+						Specifier: &xds_core.DataSource_InlineBytes{
+							InlineBytes: []byte(config.RootCert),
 						},
 					},
-					"tls_params": map[string]interface{}{
-						"tls_minimum_protocol_version": "TLSv1_2",
-						"tls_maximum_protocol_version": "TLSv1_3",
+				},
+			},
+			TlsParams: &xds_transport_sockets.TlsParameters{
+				TlsMinimumProtocolVersion: xds_transport_sockets.TlsParameters_TLSv1_2,
+				TlsMaximumProtocolVersion: xds_transport_sockets.TlsParameters_TLSv1_3,
+			},
+			TlsCertificates: []*xds_transport_sockets.TlsCertificate{
+				{
+					CertificateChain: &xds_core.DataSource{
+						Specifier: &xds_core.DataSource_InlineBytes{
+							InlineBytes: []byte(config.Cert),
+						},
 					},
-					"tls_certificates": []map[string]interface{}{
-						{
-							"certificate_chain": map[string]interface{}{
-								"inline_bytes": config.Cert,
-							},
-							"private_key": map[string]interface{}{
-								"inline_bytes": config.Key,
-							},
+					PrivateKey: &xds_core.DataSource{
+						Specifier: &xds_core.DataSource_InlineBytes{
+							InlineBytes: []byte(config.Key),
 						},
 					},
 				},
 			},
 		},
-		"load_assignment": map[string]interface{}{
-			"cluster_name": config.XDSClusterName,
-			"endpoints": []map[string]interface{}{
+	}
+	pbUpstreamTLSContext, err := ptypes.MarshalAny(upstreamTLSContext)
+	if err != nil {
+		log.Error().Err(err).Msgf("Error marshaling UpstreamTlsContext struct into an anypb.Any message")
+		return nil, err
+	}
+
+	return &xds_cluster.Cluster{
+		Name:           config.XDSClusterName,
+		ConnectTimeout: durationpb.New(time.Millisecond * 250),
```

Comment:
This was previously 25ms.

Reply:
I think it was .25 seconds which is 250 ms

Reply:
My bad, looks correct.